modules/local/pretext_graph.nf

```diff
@@ -22,15 +22,31 @@ process PRETEXT_GRAPH {
     script:
     def args         = task.ext.args ?: ''
     def prefix       = task.ext.prefix ?: "${meta.id}"
-    def PRXT_VERSION = '0.0.6'
     def UCSC_VERSION = '447' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
 
     """
-    bigWigToBedGraph ${coverage} /dev/stdout | PretextGraph ${args} -i ${pretext_file} -n "coverage" -o coverage.pretext.part
+    if [ -f "${coverage}" ];
+    then
+        bigWigToBedGraph ${coverage} /dev/stdout | PretextGraph ${args} -i ${pretext_file} -n "coverage" -o coverage.pretext.part
+    else
+        echo "SKIPPING COVERAGE"
+        mv ${pretext_file} coverage.pretext.part
+    fi
 
-    bigWigToBedGraph  ${repeat_density} /dev/stdout | PretextGraph ${args} -i coverage.pretext.part -n "repeat_density" -o repeat.pretext.part
+    if [ -f "${repeat_density}" ];
+    then
+        bigWigToBedGraph  ${repeat_density} /dev/stdout | PretextGraph ${args} -i coverage.pretext.part -n "repeat_density" -o repeat.pretext.part
+    else
+        echo "SKIPPING REPEAT_DENSITY"
+        mv coverage.pretext.part repeat.pretext.part
+    fi
 
-    bigWigToBedGraph  ${avg_coverage} /dev/stdout | PretextGraph ${args} -i repeat.pretext.part -n "avg_coverage" -o avg.pretext.part
+    if [ -f "${avg_coverage}" ]; then
```

Comment:
we are not using ${avg_coverage} anymore

Reply:
for the rest if else about bed file ingestion, could do something like: 
```
if [[ -s "${gap_file}" ]]; then
    echo "Processing GAP file..."
    cat "${gap_file}" | PretextGraph ${args} -i repeat.pretext.part -n "${gap.ft}" -o gap.pretext.part
    input_file="gap.pretext.part"
else
    input_file="repeat.pretext.part"
fi

if [[ -s "${telomere_file}" ]]; then
    echo "Processing TELO file..."
    cat "${telomere_file}" | PretextGraph ${args} -i "${input_file}" -n "${telo.ft}" -o "${prefix}.pretext"
else
    mv "${input_file}" "${prefix}.pretext"
fi

```

Reply:
Done